lib/clients/VaultClientCache.js

```diff
@@ -0,0 +1,133 @@
+     *   credentials to
+     * @param {string} filePath - path to credentials file. This file
+     *   must be a JSON file in the following format:
+     *   {"accessKey":"secretKey"}
+     * @return {VaultClientCache} this
+     */
+    loadAdminCredentialsFromFile(profileName, filePath) {
+        const adminCredsJSON = fs.readFileSync(filePath);
+        const adminCredsObj = JSON.parse(adminCredsJSON);
+        const accessKey = Object.keys(adminCredsObj)[0];
+        const secretKey = adminCredsObj[accessKey];
+
+        return this._updateProfile(profileName, {
+            adminCreds: { accessKey, secretKey },
+        });
+    }
+
+    /**
+     * Set the target Vault host name for the given profile
+     *
+     * @param {string} profileName - name of profile to bind the host
+     *   name to
+     * @param {string} host - target Vault host
+     * @return {VaultClientCache} this
+     */
+    setHost(profileName, host) {
+        return this._updateProfile(profileName, { host });
+    }
+
+    /**
+     * Set the target Vault port for the given profile
+     *
+     * @param {string} profileName - name of profile to bind the port to
+     * @param {number} port - target Vault port
+     * @return {VaultClientCache} this
+     */
+    setPort(profileName, port) {
+        return this._updateProfile(profileName, { port });
+    }
+
+    /**
+     * Set the proxy path to prepend to the URL when requesting the
+     * target Vault for the given profile
+     *
+     * @param {string} profileName - name of profile to bind the proxy
+     *   path to
+     * @param {string} proxyPath - proxy path
+     * @return {VaultClientCache} this
+     */
+    setProxyPath(profileName, proxyPath) {
+        return this._updateProfile(profileName, { proxyPath });
+    }
+
+    /**
+     * Get a VaultClient instance
+     *
+     * @param {string} profileName - name of profile used as part of
+     *   the cache key to match a VaultClient instance
+     * @param {string} [host] - host name for the target Vault,
+     *   ignored if setHost() has been called with the same profile name
+     * @param {number} [port] - port number for the target Vault,
+     *   ignored if setPort() has been called with the same profile name
+     *
+     * @return {VaultClient|null} a new or cached VaultClient instance
+     *   matching the given profile name and host/port (when
+     *   provided), or null if host or port are missing in this call
+     *   and in the profile (set through setHost()/setPort())
+     */
+    getClient(profileName, host, port) {
+        const profile = this._profiles[profileName] || {};
+        const vaultHost = profile.host || host;
+        const vaultPort = profile.port || port;
+        if (!vaultHost || !vaultPort) {
+            return null;
+        }
+        const key = `${profileName}:${vaultHost}:${vaultPort}`;
+
+        if (this._vaultClients[key] === undefined) {
+            this._vaultClients[key] = new VaultClient(
```

Comment:
Minor naming suggestion. Across all projects we use `_vaultClients` in lowercase. I recommend keeping it likewise i.e `_vaultclients`.